### PR TITLE
fix(SynchronizableRenderWindow): Fix infinitely deep recursion

### PR DESCRIPTION
**Pull request by scottwittenburg**

### CODE REVIEW

File: Sources/Rendering/Misc/SynchronizableRenderWindow/index.js [hunk 1]
```diff
@@ -225,9 +225,11 @@ export function vtkSynchronizableRenderWindow(publicAPI, model) {
 
   const addOn = createSyncFunction(publicAPI, model.synchronizerContext);
 
-  Object.keys(addOn).forEach((methodName) => {
-    publicAPI[methodName] = addOn[methodName];
-  });
+  Object.keys(addOn)
+    .filter(methodName => methodName.indexOf('SynchronizedViewId') < 0)
```
Comment (scottwittenburg) [inline Sources/Rendering/Misc/SynchronizableRenderWindow/index.js]: @jourdain  If we add the methods `getSynchronizedViewId` and `setSynchronizedViewId` to the publicAPI, then in the use case where we don't provide a `synchronizedViewId` in the initial state, we end up recursing over and over again until the call stack gets too large (see line 148 of macro.js).

I'm not sure what the actual solution is, but this fix helped me get things working at least.

Comment (jourdain) [inline Sources/Rendering/Misc/SynchronizableRenderWindow/index.js]: I would expand the `set` method in [macro](https://github.com/Kitware/vtk-js/blob/master/Sources/macro.js#L141) like below:

```
publicAPI.set = (map = {}, noWarning = false, noFunction = false) => {
    let ret = false;
    Object.keys(map).forEach((name) => {
      const fn = noFunction ? null : publicAPI[`set${capitalize(name)}`];
      if (fn && Array.isArray(map[name])) {
        ret = fn(...map[name]) || ret;
      } else if (fn) {
        ret = fn(map[name]) || ret;
      } else {
        // Set data on model directly
        if (['mtime'].indexOf(name) === -1 && !noWarning) {
          vtkWarningMacro(`Warning: Set value to model directly ${name}, ${map[name]}`);
        }
        model[name] = map[name];
        ret = true;
      }
    });
    return ret;
  };
```

Comment (jourdain) [inline Sources/Rendering/Misc/SynchronizableRenderWindow/index.js]: And then I'll keep the addOn the way it was.

**Comment by scottwittenburg** [comment]
@jourdain This could be good now, as long as you're ok with it.

**Pull request merged**